### PR TITLE
Archive functionality

### PR DESCRIPTION
**Pull request by omkarpatke**
# Added Archive Functionality 

**functionality :**
- add archive note.
- restore archive note.

**Files to review:**
- NotePage.js
- archive.js
- archive-note-context.js

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ import Login from "./Pages/Login/Login";
 import SignUp from "./Pages/SignUp/SignUp";
 import Mockman from 'mockman-js';
 import Archive from "./Pages/archivePage/archive";
+import Trash from "./Pages/Trash/trash";
+import Label from "./Pages/Label/Label";
+import Account from "./Pages/Account/Account";
+import Sidebar from "./components/Sidebar/Sidebar";
```
Comment (yaswanthmyneni) [inline src/App.js:12]: suggestion: use index file to export all pages from it in pages folder

File: src/context/archive-note-context.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+   const reStoreArchiveNote = async(note) => {
+    try {
+        const response = await axios.post(`/api/archives/restore/${note._id}` , {note} ,
+        {
+            headers: {
+                authorization: encodedToken
+            }
+        })
+        setArchiveNotes(response.data.archives);
```
Comment (yaswanthmyneni) [inline src/context/archive-note-context.js:33]: changes: check response.status before setting state.

File: src/context/note-context.js [hunk 3]
```diff
@@ -29,8 +29,22 @@ const NoteContextProvider = ({children}) => {
...
+    const addNoteToBackend = async(note) => {
+        try {
+             await axios.post('/api/notes' , {note} ,{
+                headers: {
+                    authorization: encodedToken
+                }
+            })
+        } catch (err) {
+            console.log(err);
+        }
```
Comment (yaswanthmyneni) [inline src/context/note-context.js:43]: suggestion: use console.error to display errors

File: src/Pages/archivePage/archive.js [hunk 4]
```diff
@@ -1,9 +1,33 @@
...
+    <div className="notes-container">
+      <input className='search-bar' type="search" name="search" id="search" placeholder='Search...' />
+        <h3>Archive</h3>
+        <div className="pinned-notes">
+          {allArchiveNotes.length === 0 ? "No Archive Notes" : allArchiveNotes.map((archiveNote) => (
+             <div className="note" key={archiveNote._id}>
+             <div className="title">{archiveNote.title}</div>
+             <div className="desc">{archiveNote.desc}</div>
```
Comment (yaswanthmyneni) [inline src/Pages/archivePage/archive.js:18]: suggestion: create a component for archive notes

**Review by yaswanthmyneni — APPROVED** [review]
you can merge it after changes.

**Review by Sahil25061999 — COMMENTED** [review]
The archive functionality is not working, you might need to take look

**Comment by omkarpatke** [comment]
> The archive functionality is not working, you might need to take look

but, it's working on my machine.

File: src/App.js (hunk 1 above)
Comment (omkarpatke) [reply src/App.js:12]: ok, I will work on that.

**Pull request merged**